src/plugins/discover/public/application/main/components/layout/use_fetch_more_records.test.tsx

```diff
@@ -41,90 +41,87 @@ describe('useFetchMoreRecords', () => {
   };
 
   it('should not be allowed if all records are already loaded', async () => {
+    const params = {
+      isTextBasedQuery: false,
+      stateContainer: getStateContainer({
+        fetchStatus: FetchStatus.COMPLETE,
+        loadedRecordsCount: 3,
+        totalRecordsCount: 3,
+      }),
+    };
     const {
       result: { current },
-    } = renderHook(() =>
-      useFetchMoreRecords({
-        isTextBasedQuery: false,
-        stateContainer: getStateContainer({
-          fetchStatus: FetchStatus.COMPLETE,
-          loadedRecordsCount: 3,
-          totalRecordsCount: 3,
-        }),
-      })
-    );
+    } = renderHook(() => useFetchMoreRecords(params));
```

Comment:
Yeah that's an odd one 🤔 My guess is `useObservable` calls `setState` whenever it subscribes to an observable which triggers a re-render, and since we're creating a new state container on each render, it keeps resubscribing and triggering more renders. The `fetchStatus` check in `useDataState` might have just been preventing the infinite loop.

Defining the state container outside the render call like you've done works, but if we wanted to make it a bit more explicit, we could instead use the `initialProps` option on `renderHook` to prevent recreating them on each render. Not a strong opinion though, just a suggestion:
```ts
const {
  result: { current },
} = renderHook((props: UseFetchMoreRecordsParams) => useFetchMoreRecords(props), {
  initialProps: {
    isTextBasedQuery: false,
    stateContainer: getStateContainer({
      fetchStatus: FetchStatus.COMPLETE,
      loadedRecordsCount: 3,
      totalRecordsCount: 3,
    }),
  },
});
```